lib/batch.js

```diff
@@ -145,7 +152,7 @@ internals.batch = function (batchRequest, resultsData, pos, parts, callback) {
             var ref = resultsData.resultsMap[parts[i].index];
 
             if (ref) {
-                var value = ref[parts[i].value];
+                var value = internals.resolveValue(parts[i].value, ref);
```

Comment:
I wasn't aware of it, but it sounds better than having to maintain a new function